extractor/src/main/java/org/schabi/newpipe/extractor/services/peertube/extractors/PeertubeStreamExtractor.java

```diff
@@ -402,16 +403,254 @@ private void loadSubtitles() {
+                                    .build());
+                        }
                     }
                 }
-            } catch (Exception e) {
-                // ignore all exceptions
+            } catch (final Exception ignored) {
+                // Ignore all exceptions
             }
         }
     }
 
+    private void extractLiveVideoStreams() throws ParsingException {
+        try {
+            final JsonArray streamingPlaylists = json.getArray(STREAMING_PLAYLISTS);
+            for (final Object s : streamingPlaylists) {
+                if (!(s instanceof JsonObject)) {
+                    continue;
+                }
+                final JsonObject stream = (JsonObject) s;
+                // Don't use the containsSimilarStream method because it will always return false
+                // so if there are multiples HLS URLs returned, only the first will be extracted in
+                // this case.
+                videoStreams.add(new VideoStream.Builder()
+                        .setId(String.valueOf(stream.getInt("id", -1)))
+                        .setContent(stream.getString(PLAYLIST_URL, EMPTY_STRING), true)
+                        .setIsVideoOnly(false)
+                        .setResolution(EMPTY_STRING)
+                        .setMediaFormat(MediaFormat.MPEG_4)
+                        .setDeliveryMethod(DeliveryMethod.HLS)
+                        .build());
+            }
+        } catch (final Exception e) {
+            throw new ParsingException("Could not get video streams", e);
+        }
+    }
+
+    private void getStreams() throws ParsingException {
+        // Progressive streams
+        getStreamsFromArray(json.getArray(FILES), EMPTY_STRING);
+
+        // HLS streams
+        try {
+            final JsonArray streamingPlaylists = json.getArray(STREAMING_PLAYLISTS);
+            for (final Object p : streamingPlaylists) {
+                if (!(p instanceof JsonObject)) {
+                    continue;
+                }
+                final JsonObject playlist = (JsonObject) p;
+                final String playlistUrl = playlist.getString(PLAYLIST_URL);
+                getStreamsFromArray(playlist.getArray(FILES), playlistUrl);
+            }
+        } catch (final Exception e) {
+            throw new ParsingException("Could not get streams", e);
+        }
+    }
+
+    private void getStreamsFromArray(@Nonnull final JsonArray streams,
+                                     final String playlistUrl) throws ParsingException {
+        try {
+            /*
+            Starting with version 3.4.0 of PeerTube, HLS playlist of stream resolutions contain the
+            UUID of the stream, so we can't use the same system to get HLS playlist URL of streams
+            without fetching the master playlist.
+            These UUIDs are the same that the ones returned into the fileUrl and fileDownloadUrl
+            strings.
```

Comment:
Before version 3.4.0 of PeerTube, you were able to get HLS streams by replacing the `master` part of the HLS master playlist by the height of the stream resolution (for e.g. `720` instead of `master`).

Since this version, streams are now using randoms UUIDs, so HLS URLs contains these UUIDs and this method is broken. We have now to replace the `-fragmented.` part of URLs by `.m3u8`.

So this is more a trick to get HLS streams from PeerTube without fetching the HLS manifest.